x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/inference_crud.yml

```
@@ -1214,3 +1214,40 @@ setup:
       ml.get_trained_models:
         model_id: a-regression-model-0
         include: definition_status
+
+---
+"Test include model platform architecture":
+  - do:
+      ml.put_trained_model:
+        model_id: model-without-definition
+        body: >
+          {
+            "model_type": "pytorch",
+            "inference_config": {
+              "ner": {
+              }
+            },
+          "platform_architecture": "windows-x86_64"
+          }
+
+  - do:
+      ml.get_trained_models:
+        model_id: model-without-definition
+        include: definition_status
+  - match: { count: 1 }
+  - match: { trained_model_configs.0.fully_defined: false }
+  - do:
+      ml.get_trained_models:
+        model_id: model-without-definition
+  - match: { count: 1 }
+  - match: { trained_model_configs.0.fully_defined: null }
+  - do:
+      catch: /Getting the model download status is not supported when getting more than one model/
+      ml.get_trained_models:
+        model_id: _all
+        include: definition_status
+  - do:
+      catch: /Definition status is only relevant to PyTorch model types/
+      ml.get_trained_models:
+        model_id: a-regression-model-0
+        include: definition_status
```

I don't think it's worth repeating all the checks of the test above, which is testing something different

```suggestion
  - match: { trained_model_configs.0.platform_architecture: windows-x86_64 }
```